docs/developers/components/managing.md

```diff
@@ -0,0 +1,97 @@
+# Managing Components
+
+Managing components on on Harper is easy. There are many ways to manage components with Harper, and some are limited to local development as compared to a hosted, managed Harper instance. This page will cover the recommended methods of developing, installing, deploying, and running Harper components.
+
+## Local Development
+
+Harper is designed to be simple to run locally. Generally, Harper should be installed locally on a machine using a global package manager install (i.e. `npm i -g harperdb`).
+
+> Before continuing, ensure Harper is installed and the `harperdb` CLI is available. For more information, review the [installation guide](../../deployments/install-harperdb/README.md).
+
+When developing a component locally there are a number of ways to run it on Harper.
+
+The quickest way to run a component is by using the `dev` or `run` commands (i.e. `harperdb run .`) within the component directory. Stop execution by sending a SIGINT (generally CTRL/CMD+C) signal to the process.
+
+Alternatively, to mimic interfacing with a hosted Harper instance, use operation commands instead.
+
+1. Start up Harper with `harperdb start`
+2. "Deploy" the component to the local instance by executing `harperdb deploy_component project=<name>` from within the component directory
+   1. Make sure to omit the `target` option so that it _deploys_ to the Harper instance running locally
+3. Depending on the needs of the component
+   1. If it needs a main thread restart, completely stop and start Harper again with `harperdb stop` and `harperdb start`
+   2. If it only needs a worker thread restart, run `harperdb restart` (or include `restart=true` in the deploy command from step 2)
+4. If changes need to be made to the component run `harperdb deploy_component project=<name>` again from within the component directory. It will automatically apply any changes.
```

Comment:
Should we include the `replicated` and `restart` parameters here? (we use them a lot)

Reply:
yes!